### PR TITLE
Report grade

### LINKED ISSUE #252: Judging Report button
Add the "report" button from the owner's of that project

### PR DESCRIPTION
**Pull request by ianmah**
closes #252 
image

image

### CODE REVIEW

File: src/components/Input/Button.js [hunk 1]
```diff
@@ -102,35 +98,38 @@ const StyledButton = styled.a`
...
+        ${
+          p.disabled
+            ? `cursor: not-allowed;`
+            : `background: ${hexToRgba(p.theme.colors.secondaryWarning, 0.2)};`
+        }
+      }
+      :focus {
+        box-shadow: 0 0 0 .2rem ${hexToRgba(p.theme.colors.warning, 0.5)};
+      }`}
+    ${p =>
```
Comment (jackyzha0) [inline src/components/Input/Button.js:120]: thoughts on pulling this out into a `const WarningButton = styled(StyledButton)...`? this component is getting really large

File: src/components/Judging/Submission.js [hunk 2]
```diff
@@ -50,28 +52,33 @@ const FeedbackItem = styled(Card)`
...
-const FeedbackCard = ({ feedback }) => {
+const StyledButton = styled(Button)`
+  float: right;
+`
+
+const FeedbackCard = ({ feedback, reportCallback }) => {
   return (
     <>
-      {feedback.map((item, i) => {
+      {Object.entries(feedback).map(([i, item]) => {
```
Comment (jackyzha0) [inline src/components/Judging/Submission.js:69]: why this change?

File: src/utility/firebase.js [hunk 3]
```diff
@@ -136,7 +136,27 @@ export const getUserApplication = async uuid => {
...
+        errorCallback(true)
+        return
+      }
+      const oldGrades = projectDoc.data().grades
+      const grades = { ...oldGrades, [userUid]: score }
+      transaction.update(projectsRef.doc(id), { grades })
+    })
+  } catch (e) {
+    errorCallback(true)
+    console.err(e)
```
Comment (jackyzha0) [inline src/utility/firebase.js:158]: til about console.err LOL

File: src/components/Judging/Submission.js (hunk 2 above)
Comment (ianmah) [reply src/components/Judging/Submission.js:69]: need the id to be available so we know which one was reported

File: src/components/Input/Button.js (hunk 1 above)
Comment (ianmah) [reply src/components/Input/Button.js:120]: I agree it's getting super big idk if making a new component is the best thing? Since the changes are so similar to the changes made by primary/secondary/tertiary but idk what else we can do

Comment (jackyzha0) [reply src/components/Input/Button.js:120]: do you think we can switch on p.color and simplify the if statements

**Review by jackyzha0 — APPROVED** [review]
thick component a lil sus but ill let it slide 🧞
